File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int dfs(TreeNode *root,int &ans){
        if(root==nullptr){
            return 0;
        }
        int left=dfs(root->left,ans);
        int right=dfs(root->right,ans);

        if (abs(right-left)>1){
            ans=2;
        }


        return 1+max(left,right);
        

    }
    bool isBalanced(TreeNode* root) {
        int ans=0;
        dfs(root,ans);

        return ans==2?false:true;
        
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        ans=0
        def depth(root:Optional[TreeNode],temp: int):
            nonlocal ans
            if root==None:
                ans=max(temp,ans)
                return
            depth(root.left,temp+1)
            depth(root.right,temp+1)
            return
        depth(root,0)

        return ans
            

            

        